obspy/core/inventory/util.py

```diff
@@ -840,6 +840,13 @@ def _seed_id_keyfunction(x):
         comp = "ZNE".find(x[-1])
         if comp >= 0:
             x[-1] = comp
+        # last item is component code, either the original 1-char string, or an
+        # int from 0-2 if any of "ZNE". Python3 does not allow comparison of
+        # int and string anymore (Python 2 always compares ints smaller than
+        # any string), so we need to work around this by making this last item
+        # a tuple with first item False for ints and True for strings.
+        # either a string (has `.__iter__`) or an int (doesn't have it)
+        x[-1] = (hasattr(x[-1], "__iter__"), x[-1])
```

Comment:
I meant the `hasattr`, too... It's know to be True at line 842, and not otherwise, no?